src/compiler/gcc.rs

```diff
@@ -291,7 +299,7 @@ where
 
     CompilerArguments::Ok(ParsedArguments {
         input: input.into(),
-        extension: extension,
+        extension: extension.expect("extension should always be set"),
```

Comment:
You can drop this if you rewrite the bit above to be like:
```
let extension = match extension {
  None => return CompilerArguments::CannotCache("unknown source extension"),
  Some(e) => e,
};
```

I wrote a macro to do this for required options in the Rust compiler implementation.
```